clr-installer/main.go

```diff
@@ -482,15 +597,9 @@ func execute(options args.Args) error {
 	// or we get a SIGTERM from reboot
 	signal.Reset()
 
-	if options.Reboot && installReboot {
-		_ = lock.Unlock()
-		if err := cmd.RunAndLog("reboot"); err != nil {
-			if errLog := md.Telemetry.LogRecord("reboot", 1, err.Error()); errLog != nil {
-				log.Error("Failed to log Telemetry fail record: reboot")
-			}
-			return err
-		}
-		log.RequestCrashInfo()
+	if err = processRebootOption(options, installReboot, md); err != nil {
```

Comment:
Since processRebootOption will either return an error or nil, and it is the last call, you could just simply this to:
```suggestion
	return processRebootOption(options, installReboot, md)
```

Reply:
Nice idea. Done